### PR TITLE
Blocks: move localizable strings out of the metadata attributes field

### LINKED ISSUE #36649: Blocks: attributes defined in `blocks.json` are not localized properly
### Impacted plugin

Jetpack

### Quick summary

Some Jetpack blocks (e.g., Donations, Jetpack AI Search) define their attributes in `block.json` directly. Strings defined in the `attributes` property are not localizable.

_The text_ Make a one-time donation _is defined in `block.json`_.
Screenshot 2024-03-28 at 2 10 35 PM

### Steps to reproduce

- Spin up a self-hosted Jetpack site and connect it to Jetpack
- Set the user language in another language than English
- Create a new post or page
- Add a Donations block
- Notice some texts are shown in English instead of the user language

### A clear and concise description of what you expected to happen.

All the texts should be in the user language.

### What actually happened

Some texts are not localized.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes https://github.com/Automattic/jetpack/issues/36649

## Proposed changes:

Block attributes defined in the metadata `block.json` file are not localizable (see [the WordPress.org documentation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#attributes)). This PR updates the following blocks to remove translatable strings from this file:
- Paid Content
- Jetpack AI Search
- Donations
- Subscriptions
- Cookie Consent
- Goodreads
- Mailchimp

The approach is to remove default strings from the `block.json` files, define them in a JS file instead, and use them as a fallback when reading the attributes.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Project thread: pf5801-Aw-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

For each block above, verify that it still works as expected in the editor and in the site (the Cookie Consent block may only be added in the site editor). Ideally, change the language to a language other than English and check that all UI elements are displayed in that language.

_Note: if you can't see the text in the language you chose, make sure to update the latest translations in the Update page. You can also check if the translation exists with [this tool](https://translate.wordpress.org/projects/wp-plugins/jetpack/stable/)._

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/premium-content/block.json [hunk 1]
```diff
@@ -50,22 +50,6 @@
 		"isPremiumContentChild": "isPremiumContentChild"
 	},
 	"attributes": {
-		"newPlanName": {
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/premium-content/block.json:53]: These attributes are not used anywhere any more. Removing them doesn't break backward compatibility.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php [hunk 2]
```diff
@@ -60,88 +60,6 @@ function register_block() {
 					),
 					'align'   => array( 'wide', 'full' ),
 				),
-				'attributes'      => array(
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:63]: Attributes were added back to server-side registration as a temporary workaround (https://github.com/Automattic/jetpack/pull/36695).

File: projects/plugins/jetpack/extensions/blocks/subscriptions/test/fixtures/jetpack__subscriptions__deprecated-1.serialized.html [hunk 3]
```diff
@@ -1 +1 @@
-<!-- wp:jetpack/subscriptions /-->
+<!-- wp:jetpack/subscriptions {"submitButtonText":"Subscribe"} /-->
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/test/fixtures/jetpack__subscriptions__deprecated-1.serialized.html:1]: I've followed how failing tests were fixed in previous similar situations: https://github.com/Automattic/jetpack/pull/27443.

File: projects/plugins/jetpack/extensions/blocks/mailchimp/test/fixtures/jetpack__mailchimp__deprecated-1.serialized.html [hunk 4]
```diff
@@ -1,3 +1,3 @@
-<!-- wp:jetpack/mailchimp -->
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/mailchimp/test/fixtures/jetpack__mailchimp__deprecated-1.serialized.html:1]: I've followed how failing tests were fixed in previous similar situations: https://github.com/Automattic/jetpack/pull/27443.

**Review by jeherve — COMMENTED** [review]
When adding a block, the default seems to still be the English version on my end, e.g.:

image

image

**Comment by monsieur-z** [comment]
@jeherve, updating the translations should fix this.
Screenshot 2024-04-04 at 9 13 36 AM

**Comment by jeherve** [comment]
> updating the translations should fix this.

Thanks! For some reason I had updated the translations, but needed to do once again.  ¯\_(ツ)_/¯

It works well now.

**Review by jeherve — APPROVED** [review]

**Pull request merged**